spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindSqlInjection.java

```diff
@@ -307,7 +323,50 @@ private StringAppendState updateStringAppendState(Location location, ConstantPoo
         return stringAppendState;
     }
 
-    private StringAppendState getStringAppendState(CFG cfg, ConstantPoolGen cpg) throws CFGBuilderException {
+    private StringAppendState updateJava9AndAboveStringAppendState(ClassContext ctx, Location location, ConstantPoolGen cpg,
+            StringAppendState stringAppendState) {
+        InstructionHandle handle = location.getHandle();
+        Instruction ins = handle.getInstruction();
+        if (!(ins instanceof INVOKEDYNAMIC)) {
+            throw new IllegalArgumentException("instruction must be INVOKEDYNAMIC");
+        }
+        INVOKEDYNAMIC invoke = (INVOKEDYNAMIC) ins;
+
+        ConstantPool cp = cpg.getConstantPool();
+        ConstantInvokeDynamic bmidx = (ConstantInvokeDynamic) cp.getConstant(invoke.getIndex());
+
+        JavaClass clazz = ctx.getJavaClass();
+        for (Attribute attr : clazz.getAttributes()) {
+            if (attr instanceof BootstrapMethods) {
+                BootstrapMethod bm = ((BootstrapMethods) attr).getBootstrapMethods()[bmidx.getBootstrapMethodAttrIndex()];
+                String concatArg = ((ConstantString) cp.getConstant(bm.getBootstrapArguments()[0])).getBytes(cp);
+                int u0001idx = concatArg.indexOf('\u0001');
+                if (u0001idx >= 0) {
+                    String before = concatArg.substring(0, u0001idx).trim();
+                    if (before.startsWith(",") || before.endsWith(",")) {
+                        stringAppendState.setSawComma(handle);
+                    }
+                    if (isOpenQuote(before)) {
+                        stringAppendState.setSawOpenQuote(handle);
+                    }
+
+                    String after = concatArg.substring(u0001idx + 1).trim();
+                    if (after.startsWith(",") || after.endsWith(",")) {
```

Comment:
This isn't needed if the saw comma was already set on line 347. Perhaps combine the conditions?